test/parallel/test-fs-buffertype-writesync.js

```diff
@@ -0,0 +1,19 @@
+'use strict';
+const common = require('../common');
+
+// This test insures that writeSync does support input buffers which
+// are then correctly converted into strings.
+
+const assert = require('assert');
+const fs = require('fs');
+const path = require('path');
+
+const filePath = path.join(common.tmpDir, 'test_buffer_type');
+const v = [true, false, 0, 1, Infinity, common.noop, {}, [], undefined, null];
+
+v.forEach((value) => {
+  common.refreshTmpDir();
+  const fd = fs.openSync(filePath, 'a');
```

Comment:
I would suggest moving the `common.refreshTmpDir()` out of the loop and using `'w'` here, that’s much cleaner.